torchmetrics/functional/text/wip.py

```diff
@@ -0,0 +1,88 @@
+
+from torchmetrics.functional.text.helper import _edit_distance
+
+
+def _wip_update(
+    predictions: Union[str, List[str]],
+    references: Union[str, List[str]],
+) -> Tuple[Tensor, Tensor, Tensor]:
+    """Update the wip score with the current set of references and predictions.
+
+    Args:
+        predictions: Transcription(s) to score as a string or list of strings
+        references: Reference(s) for each speech input as a string or list of strings
+    Returns:
+        Number of edit operations to get from the reference to the prediction, summed over all samples
+        Number of words overall references
+        Number of words overall prediction
+    """
+    if isinstance(predictions, str):
+        predictions = [predictions]
+    if isinstance(references, str):
+        references = [references]
+    total = tensor(0, dtype=torch.float)
+    errors = tensor(0, dtype=torch.float)
+    reference_total = tensor(0, dtype=torch.float)
+    prediction_total = tensor(0, dtype=torch.float)
+    for prediction, reference in zip(predictions, references):
+        prediction_tokens = prediction.split()
+        reference_tokens = reference.split()
+        errors += _edit_distance(prediction_tokens, reference_tokens)
+        reference_total += len(reference_tokens)
+        prediction_total += len(prediction_tokens)
+        total += max(len(reference_tokens), len(prediction_tokens))
+
+    return errors - total, reference_total, prediction_total
+
+
+def _wip_compute(errors: Tensor, reference_total: Tensor, prediction_total: Tensor) -> Tensor:
+    """Compute the Word Information Lost.
+
+    Args:
+        errors: Number of edit operations to get from the reference to the prediction, summed over all samples
+        reference_total: Number of words overall references
+        prediction_total: Number of words overall prediction
+    Returns:
+        Word Information Lost score
+    """
+    return (errors / reference_total) * (errors / prediction_total)
+
+
+def word_information_preserved(
+    predictions: Union[str, List[str]],
+    references: Union[str, List[str]],
+) -> Tensor:
+    """Word Information Lost rate is a metric of the performance of an automatic speech recognition system. This
+    value indicates the percentage of characters that were incorrectly predicted. The lower the value, the better the
+    performance of the ASR system with a Word Information Lost rate of 0 being a perfect score.
+    Args:
+        predictions: Transcription(s) to score as a string or list of strings
+        references: Reference(s) for each speech input as a string or list of strings
+    Returns:
+        Word Information Lost rate
+    Examples:
+        >>> predictions = ["this is the prediction", "there is an other sample"]
+        >>> references = ["this is the reference", "there is another one"]
+        >>> word_information_preserved(predictions=predictions, references=references)
+        tensor(0.3472)
+    """
```

Comment:
Method `word_information_preserved` contains a docstring for Word Information Lost